File: src/main/kotlin/com/contaduria/movimientofinanciero/repositories/UserRepository.java
package com.contaduria.movimientofinanciero.repositories;

import com.contaduria.movimientofinanciero.entities.FinancialUser;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

public interface UserRepository extends JpaRepository<FinancialUser, Long>, JpaSpecificationExecutor<FinancialUser> {

    Page<FinancialUser> findAll(Specification<FinancialUser> spec, Pageable pageable);


}

File: src/main/kotlin/com/contaduria/movimientofinanciero/repositories/UserRepository.kt
package com.contaduria.movimientofinanciero.repositories

import com.contaduria.movimientofinanciero.entities.User
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import org.springframework.web.bind.annotation.CrossOrigin


@CrossOrigin(origins = ["*"])
@RepositoryRestResource(path = "Users")
interface UserRepository : PagingAndSortingRepository<User?, Long?>